javaslang-test/src/main/java/javaslang/test/Gen.java

```diff
@@ -222,6 +225,19 @@
         return choose(array);
     }
 
+    static Gen<LocalDateTime> choose(LocalDateTime start, LocalDateTime end){
+        Objects.requireNonNull(start, "Start date is null");
+        Objects.requireNonNull(end, "End date is null");
+        if (end.isBefore(start)) {
+            throw new IllegalArgumentException("End date is before start date");
+        }
+        return random -> {
+            final long duration = Duration.between(start, end).toMillis();
+            final long from = Gen.choose(0, duration).apply(random);
+            return start.plus(from, ChronoUnit.MILLIS);
+        };
+    }
+
```

Comment:
I think Gen is the wrong place for this method. Gen is a general construct for specifying how to generate random values of primitive types and generic objects.

More complex logic can (and should) build solely on this.

See my next comment how to move this to Arbitrary...